### PR TITLE
feat(date-picker): add support for DatePicker to apply styles to DateInput

### LINKED ISSUE #2770: [BUG] - DatePicker cannot override Input Slot
### NextUI Version

2.3.3

### Describe the bug

DatePicker cannot override Input Slot

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

```ts
                <DatePicker 
                    name="issue_date"
                    label="Issue Date"
                    variant="bordered"
                    timeInputProps={{
                        classNames: {
                            label: "text-red-200"
                        }
                    }}
                />
```

### Expected behavior

ignore

### Screenshots or Videos

image

### Operating System Version

macOS

### Browser

Chrome

### LINKED ISSUE #2895: [BUG] - DatePicker styling classes: input and inputWrapper
### NextUI Version

2.3.6

### Describe the bug

I'm facing difficulties while trying to style the DatePicker component. I noticed that the input and inputWrapper classes are not properly passing props to the component, which is hindering proper styling. This occurs both when attempting to style directly on the component and through extendedVariants.

I've tried several approaches to style this component, but the only property I managed to use was "base". However, when attempting to style, it ends up affecting the entire container of the component, including the label and the input.

screenshot:

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

1. Use the next ui version 2.3.6 in your project.
2. Implement the DatePicker component.
3. Attempt to style the DatePicker component using the input and inputWrapper classes.
4. Notice that the input and inputWrapper classes are not properly passing props to the component, resulting in improper styling.
5. Try styling the component directly and also through extendedVariants.
6. Observe that despite trying various approaches, only the "base" property is successfully applied for styling.
7. Notice that styling with the "base" property affects the entire container of the component, including the label and the input.

### Expected behavior

I expected that when applying styles to the DatePicker component using the input and inputWrapper classes, the component would properly receive the props and apply the styling accordingly. Specifically, I anticipated that styling would be applied to the input field and its wrapper without affecting other elements within the component, such as the label.

https://github.com/nextui-org/nextui/assets/12972547/11e1b002-765e-4f02-be72-13c57f74e224

### Screenshots or Videos

with inputWrapper prop

image

with input prop
image

with base prop
image

### Operating System Version

- Windows 11

### Browser

Edge

### LINKED ISSUE #2998: DatePicker not applying inputWrapper
I want to apply a new border with color, but I can't. what is the issue?

### PR DESCRIPTION
**Pull request by ryo-manba**
Closes #2770
Closes #2895
Closes #2998 

## 📝 Description

Enabled styling the input part of DatePicker.

## ⛳️ Current behavior (updates)

Input style can't be changed.

## 🚀 New behavior

Input style can be changed.

## 💣 Is this a breaking change (Yes/No):

No.

## Summary by CodeRabbit

- **New Features**
  - Added support for applying custom styles to DateInput in the `@nextui-org/date-picker` package.

- **Documentation**
  - Updated documentation to include the new `dateInputClassNames` property, allowing users to set custom class names for date input slots.

- **Tests**
  - Expanded test suite for the `DatePicker` component to verify the correct application of custom classNames for input elements.

### CODE REVIEW

File: .changeset/cool-dolls-cheat.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@nextui-org/date-picker": minor
```
Comment (wingkwong) [inline .changeset/cool-dolls-cheat.md:2]: maybe `patch`

File: .changeset/cool-dolls-cheat.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+Add support for apply styles to DateInput (#2895)
```
Comment (wingkwong) [inline .changeset/cool-dolls-cheat.md:5]: also add `#2770` and `#2998`

**Review by wingkwong — APPROVED** [review]

**Comment by AmazingTurtle** [comment]
This is a super interesting feature as it currently completely blocks proper customization of those components. Thanks for the contribution, I hope this is merged soon! :))

File: packages/components/date-picker/src/use-date-picker.ts [hunk 3]
```diff
@@ -49,7 +49,13 @@ interface Props<T extends DateValue>
   classNames?: SlotsToClasses<DatePickerSlots> & DateInputProps<T>["classNames"];
 }
 
-export type UseDatePickerProps<T extends DateValue> = Props<T> & AriaDatePickerProps<T>;
+export type UseDatePickerProps<T extends DateValue> = Props<T> &
+  AriaDatePickerProps<T> & {
+    /**
+     * Classname or List of classes to change the classNames of the date input element.
+     */
+    dateInputClassNames?: DateInputProps<T>["classNames"];
```
Comment (jrgarciadev) [inline packages/components/date-picker/src/use-date-picker.ts:57]: @ryo-manba add this prop to the date picker documentation

**Review by jrgarciadev — APPROVED** [review]

**Pull request merged**